File: src/Restaurent.Core/Service/AddressService.cs
using Restaurent.Core.Domain.Entities;
using Restaurent.Core.Domain.RepositoryContracts;
using Restaurent.Core.DTO;
using Restaurent.Core.Helpers;
using Restaurent.Core.ServiceContracts;

namespace Restaurent.Core.Service
{
    public class AddressService : IAddressService
    {
        private readonly IAuthService _authService;
        private readonly IAddressRepository _addressRepository;

        public AddressService(IAuthService authService, IAddressRepository addressRepository)
        {
            _authService = authService;
            _addressRepository = addressRepository;
        }

        public async Task<AddressResponse> CreateAddress(AddressCreateRequest request)
        {
            if(request == null)
                throw new ArgumentNullException(nameof(request));
            ValidationHelper.ModelValidator(request);
            UserDTO? userDTO = await _authService.GetUserByUserId(request.UserId);
            if (userDTO == null)
                throw new ArgumentException("Invalid User Id");
            Address address = request.ToAddress();
            address.Id = Guid.NewGuid();
            await _addressRepository.CreateAddress(address);
            return address.ToAddressResponse();
        }

        public async Task<AddressResponse?> GetAddressByAddressId(Guid addressId)
        {
            if(addressId==Guid.Empty)
                throw new ArgumentNullException(nameof(addressId));
            Address? address =  await _addressRepository.GetAddressByAddressId(addressId);
            if (address == null)
                return null;
            return address.ToAddressResponse();
        }

        public async Task<List<AddressResponse>?> GetAddressesByUserId(Guid userId)
        {
            if (userId==Guid.Empty)
                throw new ArgumentNullException("User Id can't be null");
            UserDTO? userDTO = await _authService.GetUserByUserId(userId);
            if (userDTO == null)
                throw new ArgumentException("Invalid User Id");
            List<Address>? addresses =  await _addressRepository.GetAddressesByUserId(userId);
            if (addresses == null)
                return null;
            return addresses.Select(temp=>temp.ToAddressResponse())
                            .ToList();
        }

        public async Task<AddressResponse> UpdateAddress(AddressUpdateRequest request)
        { 
            if(request == null)
                throw new ArgumentNullException(nameof(request));
            ValidationHelper.ModelValidator(request);
            Address? matchingAddress = await _addressRepository.GetAddressByAddressId(request.AddressId);
            if (matchingAddress == null)
                throw new ArgumentException("Invalid Address Id");

            matchingAddress.Area = request.Area;
            matchingAddress.AddressLine = request.AddressLine;
            matchingAddress.City = request.City;
            matchingAddress.Landmark = request.Landmark;

            Address updatedAddress = await _addressRepository.UpdateAddress(matchingAddress);
            return updatedAddress.ToAddressResponse();
        }
    }
}

File: src/frontend/RestaurentFrontend/src/services/addressService.js
import axiosInstance from "../axios/axiosInstance"

export class Address {

    async GetAddressesOfUser(userId) {
        try {
            let response = await axiosInstance.get(`/api/Address/user-address/${userId}`)
            return response.data
        }
        catch (error) {
            logger.log("AddressService :: GetAddressesOfUser :: ",{
                status: error.response?.status,
                detail:error.response?.data?.detail || error.message
            });
            return false
        }
    }

    async GetAddressById(addressId) {
        try {
            let response = await axiosInstance.get(`/api/Address/${addressId}`)
            return response.data
        } catch (error) {
            logger.log("AddressService :: GetAddressById :: ",{
                status: error.response?.status,
                detail:error.response?.data?.detail || error.message
            });
            return false
        }
    }

    async CreateAddress({userId, addressLine, city, landmark, area}) {
        try {
            const response = await axiosInstance.post(`/api/Address`,
                {
                    UserId:userId,
                    AddressLine:addressLine,
                    City:city,
                    Landmark:landmark,
                    Area:area
                }
            );
            return response.data;
        } catch (error) {
            logger.log("AddressService :: CreateAddress :: ",{
                status: error.response?.status,
                detail:error.response?.data?.detail || error.message
            });
        }
    }

    async UpdateAddress({addressId, userId, addressLine, city, landmark, area}) {
        try {
            const response = await axiosInstance.put(`/api/Address`,{
                    AddressId:addressId,   
                    UserId:userId,
                    AddressLine:addressLine,
                    City:city,
                    Landmark:landmark,
                    Area:area
                });

            return response.data;
        } catch (error) {
            logger.log("AddressService :: UpdateAddress :: ",{
                status: error.response?.status,
                detail:error.response?.data?.detail || error.message
            });
        }
    }
}

const addressService = new Address();
export default addressService